test/unit/specs/core/createEventManager.spec.js

```diff
@@ -108,31 +108,51 @@ describe("createEventManager", () => {
         });
     });
 
-    it("sets the lastChanceCallback, which wraps config.onBeforeEventSend, on the event", () => {
-      let wrappedLastChanceCallback;
-      event.setLastChanceCallback.and.callFake(callback => {
-        wrappedLastChanceCallback = callback;
+    it("events call onBeforeEventSend callback", () => {
+      event.finalize.and.callFake(onBeforeEventSend => {
+        onBeforeEventSend();
       });
-      return eventManager.sendEvent(event, {}).then(() => {
-        wrappedLastChanceCallback();
+      return eventManager.sendEvent(event).then(() => {
         expect(config.onBeforeEventSend).toHaveBeenCalled();
       });
     });
 
-    it("logs errors in the config.onBeforeEventSend callback", () => {
-      const error = new Error("onBeforeEventSend error");
-      config.onBeforeEventSend.and.throwError(error);
+    it("returns false onBeforeEventSend and event should not be sent", () => {
+      const eventToIgnore = jasmine.createSpyObj("event", {
+        finalize: () => {},
+        shouldSend: false
+      });
+      return eventManager.sendEvent(eventToIgnore).then(() => {
+        expect(sendEdgeNetworkRequest).not.toHaveBeenCalled();
+      });
+    });
 
-      let wrappedLastChanceCallback;
-      event.setLastChanceCallback.and.callFake(callback => {
-        wrappedLastChanceCallback = callback;
+    it("returns null onBeforeEventSend and event should be sent", () => {
+      const eventToIgnore = jasmine.createSpyObj("event", {
+        finalize: () => {},
+        shouldSend: null
+      });
+      return eventManager.sendEvent(eventToIgnore).then(() => {
+        expect(sendEdgeNetworkRequest).toHaveBeenCalled();
       });
+    });
+
+    it("returns true onBeforeEventSend and event should be sent", () => {
```

Comment:
From a unit perspective, this isn't really testing what happens when `onBeforeEventSend` returns `true`. I think a more appropriate description would be: `sends when event.shouldSend returns true`. The test for whether `event.shouldSend` returns `true` when `onBeforeEventSend` returns `true` belongs in createEvent.spec.js.